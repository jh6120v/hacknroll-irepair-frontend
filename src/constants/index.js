import moment from 'moment';

export const FUNC_GO_BACK = 'GO_BACK';
export const FUNC_CLOSE = 'CLOSE';

export const CHAT_MESSAGE = 'CHAT_MESSAGE';

export const FAKE_DATA = [
    {
        id: 'A0003',
        author: '航海王',
        avatar: 'avatar-3.png',
        message: '你好啊小帥哥，連名字都可以這麼威啊～',
        images: null,
        time: +moment()
    },
    {
        id: 'A0004',
        author: '烏魯魯',
        avatar: 'avatar-4.png',
        message: '就是啊！小明，你一定有很多朋友吧！👏👏',
        images: null,
        time: +moment()
    },
    {
        id: 'A0005',
        author: 'John',
        avatar: 'avatar-5.png',
        message: '連我都忍不住想要見你了～😊',
        images: null,
        time: +moment()
    },
    {
        id: 'A0006',
        author: 'Jim',
        avatar: 'avatar-6.png',
        message: '世界很美好❤️',
        images: null,
        time: +moment()
    }
];
