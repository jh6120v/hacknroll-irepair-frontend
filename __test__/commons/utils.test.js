import { lazy } from 'react';
import { render, waitForElement } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import {
    createActionCreator,
    waitingRouteComponent,
    getRandom,
    getRandomAvatar,
    getWebSocketUrl
} from '../../src/commons/utils';
import { WEBSOCKET_URL } from '../../src/constants';

describe('test utils', () => {
    it('test createActionCreator type', () => {
        const actionCreator = createActionCreator('@@NAMESPACE');
        const test = actionCreator('TEST');

        expect(test.type).toEqual('@@NAMESPACE/TEST');
    });

    it('test createActionCreator func', () => {
        const actionCreator = createActionCreator('@@NAMESPACE');
        const test = actionCreator('TEST');

        const fakeTestCreator = () => ({
            type: '@@NAMESPACE/TEST',
            payload: {}
        });

        expect(JSON.stringify(test())).toEqual(JSON.stringify(fakeTestCreator()));
    });

    it('rendered lazily fallback', async () => {
        const LazyComponent = lazy(() => import('../lazy-component'));

        const { container } = render(
            waitingRouteComponent(LazyComponent)()
        );

        const lazyElement = await waitForElement(() => container);

        expect(lazyElement).toMatchSnapshot();
    });

    it('rendered lazily', async () => {
        const LazyComponent = lazy(() => import('../lazy-component'));

        const { getByText } = render(
            waitingRouteComponent(LazyComponent)()
        );

        const lazyElement = await waitForElement(() => getByText(/i am lazy/i));
        expect(lazyElement).toBeInTheDocument();
    });

    it('test getRandom.', () => {
        const random = getRandom(1, 3);

        expect(random).toBeGreaterThan(0);
        expect(random).toBeLessThan(4);
    });

    it('test getRandomAvatar.', () => {
        const list = [
            'avatar-1.png',
            'avatar-2.png',
            'avatar-3.png',
            'avatar-4.png',
            'avatar-5.png',
            'avatar-6.png'
        ];

        const avatar = getRandomAvatar();

        expect(list.includes(avatar)).toEqual(true);
    });

    it('test getWebSocketUrl.', () => {
        const url = getWebSocketUrl();

        expect(url).toEqual(`${WEBSOCKET_URL}`);
    });
});
