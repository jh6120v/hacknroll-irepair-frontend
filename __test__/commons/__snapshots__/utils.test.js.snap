// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test utils rendered lazily fallback 1`] = `
<div>
  <div
    class="sc-bxivhb jakkyC"
  >
    Loading...
  </div>
</div>
`;
